frontend/app.py

```diff
@@ -0,0 +1,256 @@
+    
+    return None, None
+
+def make_pv_data(site: PVSite, ts: pd.Timestamp, access_token: str = None, enphase_system_id: str = None) -> xr.Dataset:
+    live_generation_kw = None
+
+    if site.inverter_type == 'enphase' and access_token and enphase_system_id:
+        live_generation_kw = get_enphase_data(enphase_system_id, access_token)
+
+    da = process_pv_data(live_generation_kw, ts, site)
+    return da
+
+def predict_ocf(site: PVSite, model=None, ts: datetime | str = None, nwp_source: str = "icon", access_token: str = None, enphase_system_id: str = None):
+    if ts is None:
+        ts = pd.Timestamp.now().round("15min")
+    if isinstance(ts, str):
+        ts = datetime.fromisoformat(ts)
+
+    nwp_xr = get_nwp(site=site, ts=ts, nwp_source=nwp_source)
+    pv_xr = make_pv_data(site=site, ts=ts, access_token=access_token, enphase_system_id=enphase_system_id)
+
+    pred_df = forecast_v1_tilt_orientation(nwp_source, nwp_xr, pv_xr, ts, model=model)
+    return pred_df
+
+def predict_tryolabs(site: PVSite, ts: datetime | str = None):
+    solar_power_predictor = TryolabsSolarPowerPredictor()
+    
+    if ts is None:
+        start_date = pd.Timestamp.now().strftime("%Y-%m-%d")
+        start_time = pd.Timestamp.now().round(freq='h')
+    else:
+        start_date = pd.Timestamp(ts).strftime("%Y-%m-%d")
+        start_time = pd.Timestamp(ts).round(freq='h')
+
+    end_time = start_time + pd.Timedelta(hours=48)
+
+    solar_power_predictor.load_model()
+    predictions = solar_power_predictor.predict_power_output(
+        latitude=site.latitude,
+        longitude=site.longitude,
+        start_date=start_date,
+        kwp=site.capacity_kwp,
+        orientation=site.orientation,
+        tilt=site.tilt,
+    )
+
+    predictions = predictions[
+        (predictions["date"] >= start_time) & (predictions["date"] < end_time)
+    ]
+    predictions = predictions.reset_index(drop=True)
+    predictions.set_index("date", inplace=True)
+    return predictions
+    
+def run_forecast(site: PVSite, model: str = "gb", ts: datetime | str = None, nwp_source: str = "icon", access_token: str = None, enphase_system_id: str = None) -> pd.DataFrame:
+    if model == "gb":
+        return predict_ocf(site, None, ts, nwp_source, access_token, enphase_system_id)
+    elif model == "xgb":
+        return predict_tryolabs(site, ts)
+    else:  
+        raise ValueError(f"Unsupported model: {model}. Choose between 'xgb' and 'gb'")
+
+def fetch_data_and_run_forecast(access_token: str = None, enphase_system_id: str = None):
+    with st.spinner("Running forecast..."):
+        try:
+            timestamp = datetime.now().timestamp()
+            timestamp_str = datetime.fromtimestamp(timestamp, tz=timezone.utc).strftime('%Y-%m-%d %H:%M:%S')
+            ts = pd.to_datetime(timestamp_str)
+
+            site_live = PVSite(latitude=51.75, longitude=-1.25, capacity_kwp=1.25, inverter_type="enphase")
```

Comment:
I think as Peter mentioned it would be very nice if these (lat, long and capacity) could be user inputs via the UI! Although to be fair that could be in the next PR

Reply:
> I think as Peter mentioned it would be very nice if these (lat, long and capacity) could be user inputs via the UI! Although to be fair that could be in the next PR

Since it's a small feature I could add it in here too

Reply:
@aryanbhosale are you still planning on doing this in this PR?

Reply:
> @aryanbhosale are you still planning on doing this in this PR?

another PR would do I think, just so that we dont overload this one. Since the CLI version also doesnt use custom values and this PR was just to provide a UI to visualize what the CLI code did. So another PR for the custom values would make sense because it's not just limited to user inputs, we'll need to put sanity checks too